index.d.ts

```diff
@@ -0,0 +1,173 @@
+    defaultExpandedField?: string;
+    expandableDisabledField?: string;
+    selectableRows?: boolean;
+    clearSelectedRows?: boolean;
+    onRowSelected?: (row: T) => void;
+    actions?: React.ReactNode | React.ReactNode[];
+    contextTitle?: string;
+    contextActions?: React.ReactNode | React.ReactNode[];
+    noHeader?: boolean;
+    fixedHeader?: boolean;
+    fixedHeaderScrollHeight?: string;
+    subHeader?: React.ReactNode | React.ReactNode[];
+    subHeaderAlign?: string;
+    subHeaderWrap?: boolean;
+    subHeaderComponent?: React.ReactNode | React.ReactNode[];
+  }
+
+  export interface IDataTableColumn<T> {
+    name?: string;
+    selector: string;
+    sortable?: boolean;
+    format?: (row: T) => React.ReactNode;
+    cell?: (row: T) => React.ReactNode;
+    grow?: number;
+    width?: string;
+    minWidth?: string;
+    maxWidth?: string;
+    right?: boolean;
+    center?: boolean;
+    compact?: boolean;
+    ignoreOnRowClick?: boolean;
+    button?: boolean;
+    wrap?: boolean;
+    allowOverflow?: boolean;
+    hide?: number | "sm" | "md" | "lg";
+    style?: CSSProperties;
+    conditionalCellStyles?: IDataTableConditionalCellStyles<T>;
+  }
+
+  export interface IDataTableTheme {
+    title?: {
+      fontSize?: string;
+      fontColor?: string;
+      backgroundColor?: string;
+      height?: string;
+    };
+    header?: {
+      fontSize?: string;
+      fontWeight?: string;
+      fontColor?: string;
+      fontColorActive?: string;
+      backgroundColor?: string;
+      borderWidth?: string;
+      borderColor?: string;
+      borderStyle?: string;
+      height?: string;
+      denseHeight?: string;
+    };
+    contextMenu?: {
+      backgroundColor?: string;
+      fontSize?: string;
+      fontColor?: string;
+      transitionTime?: string;
+    };
+    rows?: {
+      spacing?: "default" | "spaced";
+      fontSize?: string;
+      fontColor?: string;
+      backgroundColor?: string;
+      borderStyle?: string;
+      borderWidth?: string;
+      borderColor?: string;
+      stripedColor?: string;
+      hoverFontColor?: string;
+      hoverBackgroundColor?: string;
+      height?: string;
+      denseHeight?: string;
+    };
+    cells?: {
+      cellPadding?: string;
+    };
+    expander?: {
+      fontColor?: string;
+      expanderColor?: string;
+      expanderColorDisabled?: string;
+      backgroundColor?: string;
+    };
+    pagination?: {
+      fontSize?: string;
+      fontColor?: string;
+      backgroundColor?: string;
+      buttonFontColor?: string;
+      buttonHoverBackground?: string;
+    };
+    footer?: {
+      separatorStyle?: string;
+      separatorWidth?: string;
+      separatorColor?: string;
+    };
+  }
+
+  export interface IDataTableConditionalCellStyles<T> {
+    when: (row: T) => boolean;
+    style: CSSProperties;
+  }
+
+  export interface IDataTablePaginationOptions {
+    rowsPerPageText?: string;
+    rangeSeparatorText?: string;
```

Comment:
missing `noRowsPerPage` boolean

Reply:
Fixed and added more missing properties.